dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```diff
@@ -70,14 +73,97 @@
  * - maps are written the same way as the main config dictionary <br>
  */
 public class ConfigurationStorageHelper {
-
-    public static interface ConfigStorageConsumer<K, T> {
-        T consumeConfigData(String location, Set<String> uniqueVars, Set<ConfigID> references, ConfigurationDictionary dict);
-
-        K getKey(T consumerType);
+    private static final TraceComponent tc = Tr.register(ConfigurationStorageHelper.class,
+                                                         ConfigAdminConstants.TR_GROUP, ConfigAdminConstants.NLS_PROPS);
+
+    //
+
+    /**
+     * API for receiving stored configurations.
+     *
+     * The API has two steps: Loaded configuration data consists of a location
+     * and several data structures. The location is usually a bundle location. The
+     * data structures are major subsets of configuration data: Unique variables defined
+     * by the configuration, reference IDs used by the configuration, and the table of
+     * configuration data. The first step of processing a loaded configuration is
+     * marshaling the loaded data into a configuration object. The second step of
+     * processing the loaded data is generating a key for the marshaled configuration
+     * and storing that configuration in an externally supplied table.
+     *
+     * In the current implementation, a particular value of the configuration is obtained
+     * as a particular value of the configuration data. For example, see
+     * {@link com.ibm.ws.config.admin.internal.ConfigurationStore#loadConfigurationDatas(File)},
+     * which uses the {@link Constants#SERVICE_PID} value:
+     *
+     * <code>
+     * String pid = (String) dict.get(Constants.SERVICE_PID);
+     * </code>
+     *
+     * @param <K> The type of keys used to access the loaded configurations.
+     * @param <C> The type of loaded configurations.
+     */
+    public static interface ConfigStorageConsumer<K, C> {
+        /**
+         * Consume just read configuration data.
+         *
+         * @param location The location of the configuration. Usually a bundle location.
+         * @param uniqueVars The unique variables of the configuration.
+         * @param references The reference IDs used by the configuration.
+         * @param dict The actual data of the configuration.
+         *
+         * @return The marshaled configuration data.
+         */
+        C consumeConfigData(String location,
+                            Set<String> uniqueVars, Set<ConfigID> references, ConfigurationDictionary dict);
+
+        /**
+         * Answer the key of the configuration data. This is usually a particular value of the
+         * data.
+         *
+         * @param config The configuration from which to obtain a key value.
+         * @return A key used to store the configuration.
+         */
+        K getKey(C config);
     }
 
-    private static final byte VERSION = 0;
```

Comment:
Discussed in sametime.  Decided:
*The version change should be removed.  Version is not serving the original purpose and does not need to be updated.
*The new short_string format is not necessary; string and long_string should be used.
*Because the version is not changing, the error handling code path does not need to be updated.
Changes will be made with a following commit, including test changes.